api/include/opentelemetry/nostd/list.h

```diff
@@ -0,0 +1,234 @@
+#pragma once
+
+#include <algorithm>
+#include <array>
+#include <cstring>
+#include <iterator>
+#include <iostream>
+#include <memory>
+
+#include "opentelemetry/nostd/utility.h"
+#include "opentelemetry/version.h"
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace nostd
+{
+// Non-standard implementation of the list data structure.
+template <class T>
+class list
+{
+public:
+  typedef T value_type;
+
+  // Custom iterator class
+  class iterator
+  {
+  public:
+    iterator(T *ptr) : ptr_(ptr) {}
```

Comment:
Should the ctor be private (and accessible to the friend class list)?

Reply:
Ctor moved to private.